torch/_inductor/kernel/custom_op.py

```diff
@@ -350,9 +350,72 @@ def autotune_custom_op(
     return selected_result
 
 
+def _generate_dynamic_configs(
+    tensor_inputs: list[Buffer],
+    config_generator: Callable,
+    default_impl: Callable,
+    name: str,
+) -> list[CustomOpConfig]:
+    """Generate configs dynamically based on input shapes at lowering time.
+
+    Args:
+        tensor_inputs: List of input tensors (IR Buffers)
+        config_generator: User-provided config generator function
+        default_impl: Default implementation function
+        name: Operation name for error messages
+
+    Returns:
+        List of dynamically generated CustomOpConfig objects
+
+    Raises:
+        TypeError: If config_generator returns invalid type
+        ValueError: If config_generator returns empty list
+    """
+    import inspect
+
+    from torch._inductor.virtualized import V
+
+    # Extract parameter names from custom op signature
+    sig = inspect.signature(default_impl)
+    param_names = list(sig.parameters.keys())
+
+    # Build shape dict using parameter names
+    shapes_dict = {}
+    tensor_arg_idx = 0
+    for param_name in param_names:
+        if tensor_arg_idx < len(tensor_inputs):
+            node = tensor_inputs[tensor_arg_idx]
+            raw_shape = node.get_size()
+            concrete_shape = V.graph.sizevars.size_hints(
+                raw_shape, fallback=config.unbacked_symint_fallback
+            )
+            shapes_dict[param_name] = tuple(concrete_shape)
+            tensor_arg_idx += 1
+
+    # Call user's config generator with shape dict
+    configs = config_generator(shapes_dict)
```

Comment:
why is this not passing in fake tensors ? this is inconsistent with the other api